client/normal_array_of_nillables.go

```diff
@@ -140,3 +172,41 @@ func normalizeNillableCharsArr[R string | []byte, T string | []byte](val []immut
 	}
 	return arr
 }
+
+func areNormalArraysOfNillablesEqual[T comparable](
+	val []immutable.Option[T],
+	f func() ([]immutable.Option[T], bool),
+) bool {
+	if otherVal, ok := f(); ok {
+		return areArraysOfNillablesEqual(val, otherVal)
+	}
+	return false
+}
+
+func areArraysOfNillablesEqual[T comparable](a, b []immutable.Option[T]) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i, v := range a {
+		if v != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func areArrayOfNillableBytesEqual(a, b []immutable.Option[[]byte]) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i, v := range a {
```

Comment:
todo: It looks like you may have missed a case in this loop - if only one of `v` and `b[i]` has a value it looks like this will return true.

If I'm correct, please also add a test(s) for this before fixing.

Reply:
I don't think I missed any case:
```go
if v.HasValue() && b[i].HasValue() { // if both have values, check them
    ...
} else if v.HasValue() || b[i].HasValue() { // if just one has value, return false
```

Reply:
Ah! I think I misread that as if-both-have-values, and if-neither-have-values - thanks for correcting me Islam :)